hw/ip/aon_timer/doc/_index.md

```diff
@@ -0,0 +1,104 @@
+
+The always on watchdog timer behaves similarly to the wakeup timer.
+A separate independent count starts at 0 and slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is already in low power, the wakeup signal requests the power manager to wake the system such that the IRQ can be serviced.
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which directly reboots the entire system.
+This is independent of the IRQ sent as part of the watchdog bark.
+The reset wipes out both the reset request and any pending IRQ requests, so software is not required to directly acknowledge them.
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when opeating normally.
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has 2 additional functions when compared to the always on wake timer:
+* Watchdog configuration ‘Lock’
+* Watchdog count in sleep
+
+Unlike the always on timer, the watchdog once configured, may need to be locked by firmware until the next system reset.
+This is to prevent firmware from accidentally or maliciously disabling the watchdog.
+
+The “count in sleep” option controls whether the watchdog timer continues to count in low-power mode.
+This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
+Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bite.
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
+
+See the block diagram for high level function and connectivity.
+The timer interacts with the CPU core and the pwrmgr/rstmgr to drive wakeup / reset events and interrupts.
+There is also an extra input to tell the counter whether to run (“counter-run”).
+This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a “killed” state.
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/aon_timer/data/aon_timer.hjson" >}}
+
+## Design Details
+
+The always on timer will run on a ~200KHz clock.
+The timers themselves are 32b wide, giving a maximum timeout window of roughly ~6 hours.
+
+Note however, even though the timer core runs on a slow clock, register values are sampled into the main clock domain to ensure register read / writes do not incur large latencies.
```

Comment:
This sentence feels a bit clunky. I think we could just lose the first two words, which would help.

Reply:
Rephrased